src/include/migraphx/op/select_module.hpp

```diff
@@ -57,6 +57,7 @@ struct select_module
                      param_names.cend(),
                      std::back_inserter(ret),
                      [](auto pn) { return not contains(pn, "#output_"); });
+        std::sort(ret.begin(), ret.end());
```

Comment:
Do we know exact reason why it requires sorting ? 

`get_parameter_names()` returns stable_sorted parameter list already. In other places we are using a map so order should be irrelvanat. 

Only place i see is during `compute()` of `select_module`.  `std::vector<argument> args` has fixed order. 

But `select_module` itself is added by split_single_dyn_dim pass which again uses maps and stable_sorts already. So I don't see where this problem is coming from.

Reply:
AFAIK, the parameters from `module::get_parameter_names()` are sorted by the parameter order. Which for a reason unknown to me will sometimes reorder the output parameters so they are not in name order. The  `select_module` op created by `split_single_dyn_dim` expects the output parameters to be in name order (`#output_0, #output_1, ...`) to match the order of arguments to `select_module`. The bug I would see is that a dynamic batch model would fail to run for certain batch sizes because the arguments and parameters were mismatched.